### PR TITLE
[Perl] reduce buffer copies in NDArray<->PDL conversions

### PR DESCRIPTION
**Pull request by tlby**
A bit of profiling and research lead me to try the performance test suggested in #2739 for Perl.  The performance test code
```perl
#!/usr/bin/perl
use strict;
use warnings;
use AI::MXNet 'mx';
use Benchmark;

my $a = mx->random->normal(0, 1, [ 4, 1024, 1024 ]);
mx->nd->waitall();

my $p = $a->aspdl;
my $size = length(${$p->get_dataref}) / 2 ** 30;

my $t = Benchmark::countit(-4, sub { $a->aspdl });
printf("GB/s: %g\n", $t->iters * $size / $t->[0]);
```
gave me `GB/s: 0.851562` on master.  after looking into the code a bit, this patch brings the performance test to `GB/s: 8.35156` due to reduced buffer copying overhead.

[this example](http://pdl.perl.org/?docs=API&title=PDL::API#Creating-a-piddle-manually-from-Perl) from the PDL docs uses the same pattern to load directly into the piddle without an intermediate buffer, so I think it's safe.

### CODE REVIEW

File: perl-package/AI-MXNet/lib/AI/MXNet/NDArray.pm [hunk 1]
```diff
@@ -208,17 +208,13 @@ method _sync_copyfrom(ArrayRef|PDL|PDL::Matrix $source_array)
         confess("Shape inconsistant: expected $ndary_shape_str vs got $pdl_shape_str")
     }
     my $perl_pack_type = DTYPE_MX_TO_PERL->{$dtype};
```
Comment (sergeykolychev) [inline perl-package/AI-MXNet/lib/AI/MXNet/NDArray.pm]: Please remove $perl_pack_type from the subs, as they became unused

**Comment by sergeykolychev** [comment]
nice improvement, thank you!

File: perl-package/AI-MXNet/lib/AI/MXNet/NDArray.pm (hunk 1 above)
Comment (tlby) [reply perl-package/AI-MXNet/lib/AI/MXNet/NDArray.pm]: `$perl_pack_type` is still used in deciding whether to call `_half_to_float()` or `_float_to_half()` in those subroutines.

Comment (sergeykolychev) [reply perl-package/AI-MXNet/lib/AI/MXNet/NDArray.pm]: ah yes, you're right, thanks. These blocks for half floats need to be eventually moved to AI::MXNetCAPI, they have no business to be here.

**Pull request merged**